[IMP] sales: add pricelist to loyalty prgms (#10064)
odoo/documentation: content/applications/sales/sales/products_prices/loyalty_discount.rst

   :alt: Discount and loyalty program template cards.

.. note::
   Templates **only** appear when no programs have been created, and they disappear once the first
   program is created.

Creating or editing a program opens the program form.

.. image:: loyalty_discount/price-programs.png
   :align: center
   :alt: Program options on the loyalty program form.

The program form contains the following fields:

> - :guilabel:`Program Name`: enter the name of the program. The program name is *not* visible to the
>   customer.

Reviewer: ```suggestion
- :guilabel:`Program Name`: enter the name of the program. The program name is **not** visible to the
  customer.
```

> - :guilabel:`Program Type`: select the desired :ref:`program type

Reviewer: ```suggestion
- :guilabel:`Program Type`: select the desired :ref:`program type
```

You can mention it's a drop-down menu if you want

  <sales/pricing_management/program-types>`.
- :guilabel:`Currency`: select the currency used for the program.
- :guilabel:`Pricelist`: if desired, select a pricelist from the drop-down menu to have this loyalty
  program applied to a specific pricelist (and customers attached to the pricelist). More than one
  pricelist can be selected in this field. When a single loyalty program is linked to several
  pricelists, it makes it viable for different customer segments to have different pricelists, but
  the *same* loyalty programs. If this field is left blank, the program applies to everyone,
  regardless of pricelist.
- :guilabel:`Points Unit`: enter the name of the points used for the :guilabel:`Loyalty Cards`

>   program (e.g. `Loyalty Points`). The points unit name *is* visible to the customer. This field is

Reviewer: I suggest making the 'only visible' part as an important admonition, or in it's own paragraph. I know you didn't edit that sentence but I can't help suggesting!

Author: I totally get that, and appreciate where you're coming from with the suggestion. However, just for the sake of keeping things "together" -- I think I'm going to opt to keep this wording as-is, because it quickly adds a succinct bit of information about this element, which the reader might find useful -- and I don't think it's "important enough" for its own admonition. But, as always, if someone else agrees, and thinks it should be added, I'll be sure to do so (giving you full credit for the suggestion!)

>   only available when the :guilabel:`Program Type` is set to :guilabel:`Loyalty Cards`.

Reviewer: ```suggestion
 **only** available when the :guilabel:`Program Type` is set to :guilabel:`Loyalty Cards`.
```

- :guilabel:`Start Date`: select the date on which the program becomes valid. Leave this field blank

>   if the program should always be valid and not expire.

Reviewer: ```suggestion
  if the program never expires.
```
Just for brevity.

Author: I fully appreciate this suggestion, and understand the desire for brevity. However, with this suggestion, I fear that a reader would misunderstand, and possibly think (incorrectly) that this program would *never be ABLE* to expire with this configuration in place...hence, my slightly elongated version...

just trying to make sure the reader fully understands the concept I am trying to convey. And, since we're talking about a date range (or lack thereof, in this case), I think going the "extra mile" with some additional words to explain the concept is more beneficial (though, not as brief).

- :guilabel:`End Date`: select the date on which the program stops being valid. Leave this field

>   blank if the program should always be valid and not expire.

Reviewer: ```suggestion
  blank if the program never expires.
```

Author: same reasoning as above ☝️

- :guilabel:`Limit Usage`: if desired, tick this checkbox, and enter a number of :guilabel:`usages`
  to limit the number of times the program can be used during the validity period.

> - :guilabel:`Company`: in the case of multiple companies, choose the company for which the program

Reviewer: If blank, it applies to all companies? You can add that if it is true. Can you select just 2 out of 5 for example? Or is it a one or none?

Author: I added some clarification here.

  is available.

> - :guilabel:`Available On`: select the app(s) on which the program is available.

Reviewer: ```suggestion
- :guilabel:`Available On`: select the apps on which the program is available.
```

According to the Linter/style guide, I believe we are supposed to default to the plural.

> - :guilabel:`Website`: select the website(s) on which the program is available. Leave this field

Reviewer: ```suggestion
- :guilabel:`Website`: select the websites on which the program is available. Leave this field
```

  blank to make it available on all websites.

> - :guilabel:`Point of Sale`: select the point(s) of sale at which the program is available. Leave

Reviewer: ```suggestion
- :guilabel:`Point of Sale`: select the points of sale at which the program is available. Leave
```
Points of Sale is awkward- Are these separate POS kiosks/terminals? Maybe just rephrasing it a little might sound better. Or perhaps link to a POS device doc?

Author: I understand that may come across as awkward to your ear, but it is *technically* grammatically correct. Also, given the fact that one *or multiple* Points of Sale can be selected from this field, I feel like the current way I have it formatted is the most accurate

  customer.
- :guilabel:`Buy X Get Y`: For every X item bought, the customer is granted 1 credit. After
  accumulating a specified amount of credits, the customer can trade them in to receive Y item.

> - :guilabel:`Next Order Coupons`: Generate and share single-use coupon codes that grant access to

Reviewer: ```suggestion
- :guilabel:`Next Order Coupons`: generate and share single-use coupon codes that grant access to
```
If this is a bulleted list, I believe we do the first word after the colon lowercase for all of these.
